### PR TITLE
Removed Arithmetic namespace from standard library

### PR DESCRIPTION
**Pull request by DmitryVasilevsky**
Moved ApplyXorInPlace and ApplyXorInPlaceL to Canon, and MeasureInteger to Measurement namespace.
Removed Microsoft.Quantum.Arithmetic namespace that was left empty.
The rest is a fallout.
(Also removed a few unnecessary explicit references to Microsoft.Quantum.Canon namespace, but mostly by accident.)

### CODE REVIEW

**Review by swernli — APPROVED** [review]
The changes look good to me, might be worth directly confirming with @cesarzc about kata content change and whether there is anyone else we should notify. Note that unless the katas get re-deployed they will still use the old code, which when opened in VSCode might show errors.

File: katas/content/multi_qubit_measurements/measuring_one.qs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 namespace Kata {
     open Microsoft.Quantum.Diagnostics;
-    open Microsoft.Quantum.Arithmetic;
+    open Microsoft.Quantum.Measurement;
```
Comment (swernli) [inline katas/content/multi_qubit_measurements/measuring_one.qs:3]: This change would need to be deployed to the katas site, which we should coordinate to make sure there isn't a difference between katas namespaces and what VSCode.dev supports.

File: samples/algorithms/DeutschJozsa.qs [hunk 2]
```diff
@@ -8,7 +8,6 @@
 ///
 /// This Q# program implements the Deutsch–Jozsa algorithm.
 namespace Sample {
-    open Microsoft.Quantum.Canon;
```
Comment (cesarzc) [inline samples/algorithms/DeutschJozsa.qs:11]: Was this intentional?

**Comment by cesarzc** [comment]
> The changes look good to me, might be worth directly confirming with @cesarzc about kata content change and whether there is anyone else we should notify. Note that unless the katas get re-deployed they will still use the old code, which when opened in VSCode might show errors.

We need to coordinate a deployment of this package to the Katas site, but since the package will have a lot of large changes, we should be diligent testing the new package before deploying it.

File: samples/algorithms/DeutschJozsa.qs (hunk 2 above)
Comment (DmitryVasilevsky) [reply samples/algorithms/DeutschJozsa.qs:11]: I believe Canon is open by default.

File: katas/content/multi_qubit_measurements/measuring_one.qs (hunk 1 above)
Comment (DmitryVasilevsky) [reply katas/content/multi_qubit_measurements/measuring_one.qs:3]: Cesar says we'll coordinate once this is in.

**Pull request merged**